microcosm_flask/swagger/schema.py

```diff
@@ -40,7 +45,18 @@ def build_parameter(field):
     See: https://github.com/marshmallow-code/apispec/blob/dev/apispec/ext/marshmallow/swagger.py#L81
 
     """
-    field_type, field_format = FIELD_MAPPINGS[type(field)]
+    try:
+        field_type, field_format = FIELD_MAPPINGS[type(field)]
+    except KeyError:
+        if hasattr(field, "__swagger_type__"):
+            field_type = getattr(field, "__swagger_type__")
```

Comment:
I know you preferred the string fallback; I prefer exposing more control to the user. There will be custom non-string types at some point.